src/app/components/Staking/FinalityProviders/FinalityProviderSearch.tsx

```diff
@@ -14,11 +14,19 @@ export const FinalityProviderSearch: React.FC<FinalityProviderSearchProps> = ({
   const searchParams = useSearchParams();
   const initialSearchFp = searchParams.get("fp");
 
-  const [searchTerm, setSearchTerm] = useState(initialSearchFp);
+  const [searchTerm, setSearchTerm] = useState(initialSearchFp || "");
   const debouncedSearchTerm = useDebounce(searchTerm, 300);
 
+  // Effect for handling initial search
   useEffect(() => {
-    onSearch(debouncedSearchTerm ?? "");
+    if (initialSearchFp) {
+      onSearch(initialSearchFp);
+    }
+  }, [initialSearchFp, onSearch]);
+
+  // Effect for handling subsequent searches
+  useEffect(() => {
+    onSearch(debouncedSearchTerm);
   }, [debouncedSearchTerm, onSearch]);
```

Comment:
What issue we're trying to fix? Previous implementation works for me on my local

Reply:
It doesn't work on production environment